### PR TITLE
Roll cycle bug

### PR DESCRIPTION
**Pull request by cmunger**
Providing you an unit test that demonstrates a problem with files roll cycle and tailer when tailer is in direction none and a roll cycle occurs. Forget commit 2, it just contains a sysout log that print when the error occurs in the SingleChronicleQueueExcerpts. I had no time to really investigate the bug since it's too complicated for me to debug. This problem is stopping the adoption of your library in our company so I would really love to see it fixed. We discovered this problem when implementing a java Queue interface over a chronicle queue. The peek, pool, skip, offer methods in the test reflects how we have implemented things in our java queue interface impl of over chronicle. I hope i'm doing nothing anything wrong with the chronicle api usage. thanks

### CODE REVIEW

**Comment by cmunger** [comment]
Forgot to mention that the test have been done with the latest version of chronicle (2.6.41-SNAPSHOT)

**Comment by dpisklov** [comment]
@cmunger 4.6.412-SNAPSHOT I assume?

**Comment by cmunger** [comment]
yep 4.6.41 sorry :)

File: src/test/java/net/openhft/chronicle/queue/ChronicleRollingIssueWithDirectionsTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+import net.openhft.chronicle.bytes.Bytes;
+import net.openhft.chronicle.bytes.HeapBytesStore;
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.core.OS;
+import net.openhft.chronicle.wire.DocumentContext;
+
+/**
+ * This tests demonstrates that when "peeking" data (using tailer with direction none), if we peek data over a roll cycle, the returned data is
+ * not at the right index when teh chronicle file is rolled and calling tailer.readingDocument() the returned index is the same twice. The
```
Comment (MartyIX) [inline src/test/java/net/openhft/chronicle/queue/ChronicleRollingIssueWithDirectionsTest.java]: teh -> the

**Comment by annaftarrant** [comment]
Would you be interested in paid support to help you resolve this issue?

**Comment by RobAustin** [comment]
@epickrram could you merge this or just close it

**Comment by cmunger** [comment]
Hello,

Sorry guys but i cannot pay support for that. it took me already quite a lot of time to figure out this problem, bug is still present with version 4.6.59. Tests case still demonstrate it.

Happy new year :)

**Comment by RobAustin** [comment]
@cmunger Happy new year, Sorry for the delay in replying to you, we are vary keen to take your pull request but the problem I'm having is that our code as moved on somewhat from your original pull request, The changes are so huge that I'm not able to confidently manually merge your changes. I'd very much appreciate it if you could submit your pull request again ( based on the last master code base ). As we have huge conflicts in 
```
src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueExcerpts.java
```

**Comment by cmunger** [comment]
Yeah sorry it was my first pull request and the file you mention should not have been included, only the unit test is valid and demonstrates the 'problem'. So do not merge src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueExcerpts.java

**Comment by RobAustin** [comment]
please - resubmit the pull request, thanks

**Comment by epickrram** [comment]
I've pulled the unit test into master and added bug #417

**Comment by epickrram** [comment]
@cmunger see #417

**Pull request closed without merging**